### PR TITLE
Feature/compose

### PR DESCRIPTION
**Pull request by sierraobryan**
# Changelog
Moved the app over to compose. Removed old fragments and navigation structure

## Added
- compose

## Changed
- navigation

## Removed
- Old navigation and fragments

## Is there test coverage for your changes?
  Unit Tests: Yes
  Espresso Tests: NO

### CODE REVIEW

File: app/src/main/java/com/atomicrobot/carbon/StartActivity.kt [hunk 1]
```diff
@@ -3,27 +3,34 @@ package com.atomicrobot.carbon
...
     private val splashViewModel: SplashViewModel by viewModels()
-    public override fun onCreate(savedInstanceState: Bundle?) {
+    override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        setContentView(R.layout.activity_start)
-
+        setContent {
+            CarbonAndroidTheme {
+                MainNavigation(true)
```
Comment (Brandon-Wolff) [inline app/src/main/java/com/atomicrobot/carbon/StartActivity.kt:23]: This should be passing in the return value from `handleIntent()`

File: app/src/main/java/com/atomicrobot/carbon/StartActivity.kt [hunk 2]
```diff
@@ -3,27 +3,34 @@ package com.atomicrobot.carbon
...
+            }
+        }
         handleIntent(intent)
     }
 
-    private fun handleIntent(intent: Intent) {
+    private fun handleIntent(intent: Intent): Boolean {
+        Timber.d("handle intent")
```
Comment (Brandon-Wolff) [inline app/src/main/java/com/atomicrobot/carbon/StartActivity.kt:30]: Do we want this log statement here?

File: app/src/main/java/com/atomicrobot/carbon/ui/BaseActivity.kt [hunk 3]
```diff
@@ -1,10 +1,6 @@
 package com.atomicrobot.carbon.ui
 
 import androidx.appcompat.app.AppCompatActivity
-import androidx.navigation.Navigation
-import com.atomicrobot.carbon.R
 
 abstract class BaseActivity : AppCompatActivity() {
```
Comment (Brandon-Wolff) [inline app/src/main/java/com/atomicrobot/carbon/ui/BaseActivity.kt:5]: This file can be deleted

File: app/src/main/java/com/atomicrobot/carbon/ui/compose/MainNavigation.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                if (isDeepLinkIntent) {
+                    navController.navigate(viewModel.getDeepLinkNavDestination())
+                }
+                else {
+                    navController.navigate(StartActivity.mainPage)
+                }
+            }
+
+        }
+        composable(StartActivity.mainPage) { Main(mainViewModel) }
```
Comment (Brandon-Wolff) [inline app/src/main/java/com/atomicrobot/carbon/ui/compose/MainNavigation.kt:35]: Do we want the main screen to just be called `Main()`? On the koin branch I called it `MainScreen` to match with `SplashScreen` etc.

Comment (atomicrobotjohn) [reply app/src/main/java/com/atomicrobot/carbon/ui/compose/MainNavigation.kt:35]: Adding "Screen" sounds like a good way to go

File: app/src/main/java/com/atomicrobot/carbon/ui/BaseActivity.kt (hunk 3 above)
Comment (atomicrobotjohn) [reply app/src/main/java/com/atomicrobot/carbon/ui/BaseActivity.kt]: 👍

File: app/src/main/java/com/atomicrobot/carbon/StartActivity.kt (hunk 2 above)
Comment (atomicrobotjohn) [reply app/src/main/java/com/atomicrobot/carbon/StartActivity.kt:30]: removed

File: app/src/main/java/com/atomicrobot/carbon/StartActivity.kt (hunk 1 above)
Comment (atomicrobotjohn) [reply app/src/main/java/com/atomicrobot/carbon/StartActivity.kt:23]: nice catch

**Review by ascheetz — APPROVED** [review]
So many ktlint changes

**Pull request merged**